packages/core/src/utils/generators.test.ts

```diff
@@ -74,7 +74,7 @@ test("mergeAsyncGenerators() yields all results", async () => {
 
   await new Promise((res) => setTimeout(res));
 
-  expect(results).toStrictEqual([1, 3, 2, 4]);
+  expect(results).toStrictEqual([1, 2, 3, 4]);
```

Comment:
What happened here?

Reply:
I changed the `mergeAsyncGenerators` function to "look ahead" by one.

https://github.com/ponder-sh/ponder/blob/3ece6520bdf0f9dac8c4b68a0df45337d20d2abe/packages/core/src/utils/generators.ts#L26-L27